utils/model_uploader/model_autotracing.py

```diff
@@ -0,0 +1,596 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+# This program is run by "Model Auto-tracing & Uploading" workflow
+# (See model_uploader.yml) to perform model auto-tracing and prepare
+# files for uploading to OpenSearch model hub.
+
+import argparse
+import json
+import os
+import shutil
+import sys
+import warnings
+from typing import List, Optional, Tuple
+from zipfile import ZipFile
+
+import numpy as np
+from mdutils.fileutils import MarkDownFile
+from numpy.typing import DTypeLike
+from sentence_transformers import SentenceTransformer
+
+# We need to append ROOT_DIR path so that we can import
+# OPENSEARCH_TEST_CLIENT and opensearch_py_ml since this
+# python script is not in the root directory.
+THIS_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.join(THIS_DIR, "../..")
+sys.path.append(ROOT_DIR)
+
+LICENSE_PATH = "LICENSE"
+from opensearch_py_ml.ml_commons import MLCommonClient
+from opensearch_py_ml.ml_models.sentencetransformermodel import SentenceTransformerModel
+from tests import OPENSEARCH_TEST_CLIENT
+
+BOTH_FORMAT = "BOTH"
+TORCH_SCRIPT_FORMAT = "TORCH_SCRIPT"
+ONNX_FORMAT = "ONNX"
+
+TEMP_MODEL_PATH = "temp_model_path"
+TORCHSCRIPT_FOLDER_PATH = "model-torchscript/"
+ONNX_FOLDER_PATH = "model-onnx/"
+UPLOAD_FOLDER_PATH = "upload/"
+MODEL_CONFIG_FILE_NAME = "ml-commons_model_config.json"
+OUTPUT_DIR = "trace_output/"
+LICENSE_VAR_FILE = "apache_verified.txt"
+DESCRIPTION_VAR_FILE = "description.txt"
+TEST_SENTENCES = [
+    "First test sentence",
+    "This is a very long sentence used for testing model embedding outputs.",
+]
+RTOL_TEST = 1e-03
+ATOL_TEST = 1e-05
+ML_BASE_URI = "/_plugins/_ml"
+
+
+def verify_license_in_md_file() -> bool:
```

Comment:
I think we should support other open source license in future. Maybe add some TODO to support more license.

Reply:
Added